### PR TITLE
Fix build issues

### PR DESCRIPTION
**Pull request by thornbill**
Below is a summary of the changes made:
* Add package.json for dependencies
* Fix issues with new clean-css and uglify-js versions
* Update README.md to include build instructions

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,50 @@
...
+   ```sh
+   npx cordova prepare
+   ```
+
+### Android
+
+#### Prerequisites
+
+* [Android Studio](https://developer.android.com/studio/install)
+* Android SDK Build-Tools 23.0.2
```
[the remark this answers is not in the document]
Comment (thornbill) [reply README.md]: * Android SDK Build-Tools 23.0.2
  * This is specified in the [project.properties](https://github.com/jellyfin/jellyfin-android/blob/master/platforms/android/project.properties#L13). I'm not sure if we could change that version without updating the Android platform. Multiple versions of these tools can be installed via the SDK manager.
* Android SDK Tools 25.2.3
  * Newer versions of the Android SDK Tools changed the location of some of the build tools. These changes aren't supported by the modified version of the Android platform used here.

**Comment by nvllsvm** [comment]
Doesn't work for me:

```
$ npm --version
6.5.0
$ npm install
npm WARN jellyfin-android@2.8.54 No repository field.

audited 2804 packages in 2.479s
found 3 vulnerabilities (1 low, 2 high)
  run `npm audit fix` to fix them, or `npm audit` for details
$ npx cordova prepare
Unable to load PlatformApi from platform. Error: Cannot find module 'q'
Unhandled error. (The platform "android" does not appear to be a valid cordova platform. It is missing API.js. android not supported.)
```

**Comment by thornbill** [comment]
@nvllsvm I encountered that issue when running newer versions of cordova. Could you verify the version you have installed matches what is in the `package.json`?
```
➜ npx cordova --version
6.5.0
```
I just verified a clean install works for me on a different machine.

**Comment by nvllsvm** [comment]
Ahh, I do have a newer version.

```
$ npx cordova --version
8.1.2 (cordova-lib@8.1.1)
```

Is there a way to pin to an older version? I'm not familiar with Node tools.

**Comment by nvllsvm** [comment]
I'm a dumbass - was using your `build-issues` branch instead of `fix-build-issues`.

# Issue 1 (Fixed)
After switchin branches, I needed to bump the version of Cordova else I got a build error

Error:
```
$ npx cordova run android --debug
...
Skipping minification for /home/ar/jellyfin-android/platforms/android/assets/www/bower_components/jstree/dist/themes/default/style.min.css
encountered a .gif file, not compressing it
minifying css file /home/ar/jellyfin-android/platforms/android/assets/www/bower_components/emby-webcomponents/sharing/social-share-kit-1.0.10/dist/css/social-share-kit.css
encountered a .eot file, not compressing it
encountered a .svg file, not compressing it
encountered a .ttf file, not compressing it
encountered a .woff file, not compressing it
uglifying js file /home/ar/jellyfin-android/platforms/android/assets/www/bower_components/emby-webcomponents/sharing/social-share-kit-1.0.10/dist/js/social-share-kit.js
Skipping minification for /home/ar/jellyfin-android/platforms/android/assets/www/bower_components/emby-webcomponents/sharing/social-share-kit-1.0.10/dist/js/social-share-kit.min.js
ANDROID_HOME=/home/ar/.android/sdk
JAVA_HOME=/opt/android-studio/jre
Error: Could not find gradle wrapper within Android SDK. Might need to update your Android SDK.
Looked here: /home/ar/.android/sdk/tools/templates/gradle/wrapper

```
Fix
```
$ npx cordova platform update android@6.2.2
```

# Issue 2 (Not Fixed)

Then I manually installed Gradle from the Arch Linux repos. Build then failed.
```
$ npx cordova run android   
...
minifying css file /home/ar/jellyfin-android/platforms/android/assets/www/bower_components/jstree/dist/themes/default-dark/style.css
Skipping minification for /home/ar/jellyfin-android/platforms/android/assets/www/bower_components/jstree/dist/themes/default-dark/style.min.css
encountered a .gif file, not compressing it
uglifying js file /home/ar/jellyfin-android/platforms/android/assets/www/bower_components/jquery/external/sizzle/dist/sizzle.js
Skipping minification for /home/ar/jellyfin-android/platforms/android/assets/www/bower_components/jquery/external/sizzle/dist/sizzle.min.js
minifying css file /home/ar/jellyfin-android/platforms/android/assets/www/bower_components/emby-webcomponents/sharing/social-share-kit-1.0.10/dist/css/social-share-kit.css
uglifying js file /home/ar/jellyfin-android/platforms/android/assets/www/bower_components/emby-webcomponents/sharing/social-share-kit-1.0.10/dist/js/social-share-kit.js
Skipping minification for /home/ar/jellyfin-android/platforms/android/assets/www/bower_components/emby-webcomponents/sharing/social-share-kit-1.0.10/dist/js/social-share-kit.min.js
encountered a .eot file, not compressing it
encountered a .svg file, not compressing it
encountered a .ttf file, not compressing it
encountered a .woff file, not compressing it
ANDROID_HOME=/home/ar/.android/sdk
JAVA_HOME=/opt/android-studio/jre
Subproject Path: CordovaLib
null
The Task.leftShift(Closure) method has been deprecated and is scheduled to be removed in Gradle 5.0. Please use Task.doLast(Action) instead.
	at build_339y134hcp2n0ir9pagq0rtjk.run(/home/ar/jellyfin-android/platforms/android/build.gradle:138)
org.xwalk:xwalk_core_library:21+
The JavaCompile.setDependencyCacheDir() method has been deprecated and is scheduled to be removed in Gradle 4.0.
Incremental java compilation is an incubating feature.
The TaskInputs.source(Object) method has been deprecated and is scheduled to be removed in Gradle 4.0. Please use TaskInputs.file(Object).skipWhenEmpty() instead.

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'android'.
> java.lang.NullPointerException (no error message)

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 0.519 secs
Error: /home/ar/jellyfin-android/platforms/android/gradlew: Command failed with exit code 1 Error output:
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'android'.
> java.lang.NullPointerException (no error message)

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
```

**Comment by nvllsvm** [comment]
Fixed my issue by downgrading to SDK Tools 25.2.3.
http://dl-ssl.google.com/android/repository/tools_r25.2.3-linux.zip

**Pull request merged**